src/content_handlers/apt_handler/src/apt_parser.cpp

```diff
@@ -88,6 +92,18 @@ std::unique_ptr<AptContent> GetAptContentFromRootValue(JSON_Value* rootValue)
                     ADUC_ERC_APT_HANDLER_INVALID_PACKAGE_DATA);
             }
 
+            // Note: For backward compatibility.
+            //
+            // Are we installing adu-agent package?
+            // Currently, we are assuming adu-agent* or du-agent* is an ADU Agent package.
+            if (!aptContent->AgentRestartRequired
+                && ((name.find_first_of("adu-agent") == 0) || (name.find_first_of("deviceupdate-agent") == 0)))
+            {
+                aptContent->AgentRestartRequired = true;
```

Comment:
>aptContent->AgentRestartRequired = true; [](start = 16, length = 40)

is this always true?

Reply:
only if package's name (in APT manifest) implies that we're updating the agent.

---
In reply to: [595997484](https://github.com/Azure/iot-hub-device-update/pull/17#discussion_r595997484) [](ancestors = 595997484)

Reply:
not blocking at  the moment, but are we sure that we MUST reboot after an deviceupdate-agent package install? Any way to avoid that? reboots stink.

---
In reply to: [596352894](https://github.com/Azure/iot-hub-device-update/pull/17#discussion_r596352894) [](ancestors = 596352894,595997484)